Classes/Core/Functional/Framework/DataHandling/ActionService.php

```diff
@@ -491,10 +491,6 @@ protected function getVersionedId(string $tableName, $liveUid)
             ->select('uid')
             ->from($tableName)
             ->where(
-                $queryBuilder->expr()->eq(
-                    'pid',
-                    $queryBuilder->createNamedParameter(-1, \PDO::PARAM_INT)
```

Comment:
Probably might select discarded version records... which should not be resolved as "live version"

Reply:
Please adjust restrictions like this

```
$queryBuilder->getRestrictions()->removeAll()->add(GeneralUtility::makeInstance(DeletedRestriction::class));
```